metrics/impl/default_metric_manager.go

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package impl
+
+import (
+	"sync"
+)
+
+import (
+	"github.com/apache/dubbo-go/common/constant"
+	"github.com/apache/dubbo-go/common/extension"
+	"github.com/apache/dubbo-go/config"
+	"github.com/apache/dubbo-go/metrics"
+)
+
+func init() {
+	extension.SetMetricManager(constant.DEFAULT_KEY, newDefaultMetricManager())
+}
+
+/*
+ * this implementation dependent on  the MetricRegistry.
+ * com.alibaba.metrics.AliMetricManager
+ */
+type DefaultMetricManager struct {
+	// group name -> MetricRegistry
+	metricRegistryMap sync.Map
```

Comment:
I think it is better to maintain with normal map and lock

Reply:
actually i agree to use sync.Map,   this scene is fit to sync.Map use case,i think metricRegistryMap will be a lot of data

>>The Map type is optimized for two common use cases: (1) when the entry for a given key is only ever written once but read many times, as in caches that only grow, or (2) when multiple goroutines read, write, and overwrite entries for disjoint sets of keys. In these two cases, use of a Map may significantly reduce lock contention compared to a Go map paired with a separate Mutex or RWMutex.

Ref: [sync.Map godoc](https://golang.org/pkg/sync/#Map)

Reply:
> actually i agree to use sync.Map, this scene is fit to sync.Map use case,i think metricRegistryMap will be a lot of data
> 
> > > The Map type is optimized for two common use cases: (1) when the entry for a given key is only ever written once but read many times, as in caches that only grow, or (2) when multiple goroutines read, write, and overwrite entries for disjoint sets of keys. In these two cases, use of a Map may significantly reduce lock contention compared to a Go map paired with a separate Mutex or RWMutex.
> 
> Ref: [sync.Map godoc](https://golang.org/pkg/sync/#Map)

get it, thx